Paymetheus/ViewModels/PurchaseTicketsViewModel.cs

```diff
@@ -0,0 +1,404 @@
+                    return;
+                }
+            }
+
+            _purchaseTickets.Executable = true;
+        }
+
+        private StakeDifficultyProperties _stakeDifficultyProperties;
+        public StakeDifficultyProperties StakeDifficultyProperties
+        {
+            get { return _stakeDifficultyProperties; }
+            internal set { _stakeDifficultyProperties = value; RaisePropertyChanged(); }
+        }
+
+        public ICommand FetchStakeDifficultyCommand { get; }
+
+        private async void FetchStakeDifficultyAction()
+        {
+            try
+            {
+                StakeDifficultyProperties = await App.Current.Synchronizer.WalletRpcClient.StakeDifficultyAsync();
+                int _windowSize = 144;
+                int _heightOfChange = ((StakeDifficultyProperties.Height / _windowSize) + 1) * _windowSize;
+                BlocksToRetarget = _heightOfChange - StakeDifficultyProperties.Height;
+            }
+            catch (Exception ex)
+            {
+                MessageBox.Show(ex.Message, "Error");
+            }
+        }
+
+        private int _blocksToRetarget;
+        public int BlocksToRetarget
+        {
+            get { return _blocksToRetarget; }
+            internal set { _blocksToRetarget = value; RaisePropertyChanged(); }
+        }
+
+        private void PurchaseTicketsAction()
+        {
+            var shell = ViewModelLocator.ShellViewModel as ShellViewModel;
+            if (shell != null)
+            {
+                var _account = SelectedAccount.Account;
+                var _spendLimit = StakeDifficultyProperties.Price;
+                int _requiredConfirms = 2; // TODO allow user to set
+                uint _expiryHeight = (uint)_expiry + (uint)StakeDifficultyProperties.Height;
+
+                Amount _splitFeeLocal = 0;
+                Amount _ticketFeeLocal = 0;
+                if (_feesChecked)
+                {
+                    _splitFeeLocal = _splitFee;
+                    _ticketFeeLocal = _ticketFee;
+                }
+
+                Func<string, Task<bool>> action =
+                    passphrase => PurchaseTicketsWithPassphrase(passphrase, _account, _spendLimit, 
+                    _requiredConfirms, _ticketAddress, (uint)_number, _poolAddress, _poolFees,
+                    _expiryHeight, _splitFeeLocal, _ticketFeeLocal);
+                shell.VisibleDialogContent = new PassphraseDialogViewModel(shell, 
+                    "Enter passphrase to purchase tickets", 
+                    "PURCHASE", 
+                    action);
+            }
+        }
+
+        private string _responseString = "";
+        public string ResponseString
+        {
+            get { return _responseString; }
+            set { _responseString = value; RaisePropertyChanged(); }
+        }
+
+        private async Task<bool> PurchaseTicketsWithPassphrase(string passphrase, Account account, 
```

Comment:
Almost all of these parameters are already class members and don't need to be passed in.  The passphrase is the only one that isn't know by the class (because it is passed in by the dialog).